src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Serialization/ObjectWriters.cs

```diff
@@ -92,6 +92,32 @@ public static void WriteProperties(JsonDataWriter writer, TagHelperDescriptor va
         writer.WriteArrayIfNotNullOrEmpty(nameof(value.Diagnostics), value.Diagnostics, Write);
         writer.WriteObject(nameof(value.Metadata), value.Metadata, WriteMetadata);
 
+        static void WriteDocumentationObject(JsonDataWriter writer, string propertyName, object? documentationObject)
+        {
+            switch (documentationObject)
+            {
+                case DocumentationDescriptor descriptor:
+                    writer.WriteObject(propertyName, descriptor, static (writer, value) =>
+                    {
+                        writer.Write(nameof(value.Id), (int)value.Id);
+                        if (value.Args is { Length: > 0 })
+                        {
+                            writer.WriteArray(nameof(value.Args), value.Args, static (w, v) => w.WriteValue(v));
```

Comment:
Nit: could you put a parameter name on the first arg? I needed to go double-check what this version of `WriteArray` does, since it's not just "write an array", it's "write a property name and then an array".

Reply:
If I add it here, I would expect it to be added to all of the `Write*` calls above and throughout -- hundreds of calls. That feels excessive to me. Do you really feel this is necessary? Personally, it seems excessive and the meaning of the code isn't all that unclear if you're familiar with the pattern used consistently throughout. What do you think?

Reply:
That's fair. The other instanced I'd seen used a parameter that was more specifically named `propertyName`, but I only looked at a couple; if there are more throughout the type of this form, then I agree that's probably too big a refactoring for this pr.

Reply:
FWIW, I will definitely take this as API feedback though. Even though this is an internal API for reading/writing data, I may be going a bit overboard on overloads (see what I did there?). I'll look at updating this API in an upcoming PR to be clearer.